// @everreal/er-common-helpers/lib/errors/__tests__/errorMappers.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var errorMappers_1 = require("./../errorMappers");
describe('WHEN errorMappers', function () {
    describe('WHEN calling mapErrorToJSON with a swagger validation error', function () {
        var swaggerError = new Error();
        // Simulate a swagger error
        swaggerError.code = 'SCHEMA_VALIDATION_FAILED';
        swaggerError.results = {
            errors: [
                {
                    SOME_ERROR_PROP: 'SOME_ERROR_PROP',
                },
            ],
        };
        var mappedError;
        var mockRequest = {
            body: {
                test: true,
            },
        };
        beforeEach(function () { return (mappedError = errorMappers_1.mapErrorToJSON(swaggerError, mockRequest)); });
        it('SHOULD return correct mapped error', function () {
            expect(mappedError.message.toLowerCase()).toContain('swagger');
            expect(mappedError.errors[0]).toBeDefined();
            expect(mappedError.errors[0].SOME_ERROR_PROP).toBeDefined();
        });
    });
});
