// @everreal/er-common-helpers/lib/errors/__mocks__/errors.mock.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.swaggerSchemaValidationFailedError = {
    code: 'SCHEMA_VALIDATION_FAILED',
    failedValidation: true,
    results: {
        errors: [
            {
                code: 'INVALID_TYPE',
                message: 'Expected type string but found type object',
                path: ['contractDetails', 'petsAllowed'],
                description: 'Are pets allowed.',
            },
            {
                code: 'ENUM_MISMATCH',
                message: 'No enum match for: {"YES":"YES","NO":"NO","DEPENDS":"DEPENDS"}',
                path: ['contractDetails', 'petsAllowed'],
                description: 'Are pets allowed.',
            },
        ],
        warnings: [],
    },
    path: ['paths', '/listing', 'post', 'parameters', '0'],
    paramName: 'body',
};
exports.swaggerRequiredError = {
    code: 'REQUIRED',
    failedValidation: true,
    path: ['paths', '/listing', 'post', 'parameters', '3'],
    paramName: 'skip',
};
exports.swaggerMissingRequiredPropertyError = {
    code: 'SCHEMA_VALIDATION_FAILED',
    failedValidation: true,
    results: {
        errors: [
            { code: 'OBJECT_MISSING_REQUIRED_PROPERTY', message: 'Missing required property: period', path: ['contractDetails'] },
        ],
        warnings: [],
    },
    path: ['paths', '/listing/rent/short-term-apartment', 'post', 'parameters', '0'],
    paramName: 'body',
};
exports.sequelizeError = {
    name: 'SequelizeUniqueConstraintError',
    message: 'Validation error',
    errors: [
        {
            message: 'listing_created_by_company_id must be unique',
            type: 'unique violation',
            path: 'listing_created_by_company_id',
            value: 'bc7d6200-f873-11e6-8f7f-6312d0182b41',
        },
        {
            message: 'title must be unique',
            type: 'unique violation',
            path: 'title',
            value: 'Listing 1',
        },
    ],
    fields: {
        listing_created_by_company_id: 'bc7d6200-f873-11e6-8f7f-6312d0182b41',
        title: 'Listing 1',
    },
    parent: {
        name: 'error',
        length: 339,
        severity: 'ERROR',
        code: '23505',
        detail: 'Key (listing_created_by_company_id, title)=(bc7d6200-f873-11e6-8f7f-6312d0182b41, Listing 1) already exists.',
        schema: 'public',
        table: 'listings',
        constraint: 'listings_listing_created_by_company_id_title_key',
        file: 'nbtinsert.c',
        line: '433',
        routine: '_bt_check_unique',
    },
    original: {
        name: 'error',
        length: 339,
        severity: 'ERROR',
        code: '23505',
        detail: 'Key (listing_created_by_company_id, title)=(bc7d6200-f873-11e6-8f7f-6312d0182b41, Listing 1) already exists.',
        schema: 'public',
        table: 'listings',
        constraint: 'listings_listing_created_by_company_id_title_key',
        file: 'nbtinsert.c',
        line: '433',
        routine: '_bt_check_unique',
    },
};
