// @everreal/er-common-helpers/lib/services/languageService/languageService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var cookieService = require("../cookieService/cookieService");
var constants_1 = require("./../../helpers/constants");
function getCurrentLanguage(params) {
    if (params === void 0) { params = {}; }
    var req = params.req, _a = params.defaultLanguage, defaultLanguage = _a === void 0 ? constants_1.LANGUAGES.__DEFAULT : _a;
    return cookieService.getCookie({ req: req, name: 'accept-language' }) || defaultLanguage;
}
exports.getCurrentLanguage = getCurrentLanguage;
function setCurrentLanguage(params) {
    if (params === void 0) { params = {}; }
    var res = params.res, path = params.path, language = params.language, _a = params.domain, domain = _a === void 0 ? '.everreal.co' : _a;
    cookieService.setCookie({
        res: res,
        domain: domain,
        path: path,
        name: 'accept-language',
        value: language,
        maxAge: 60 * 60 * 24 * 365,
    });
}
exports.setCurrentLanguage = setCurrentLanguage;
