// @everreal/er-common-helpers/lib/helpers/constants/applicationForm.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var common_1 = require("./common");
var APPLICATION_FORM_FIELD_STATUS;
(function (APPLICATION_FORM_FIELD_STATUS) {
    APPLICATION_FORM_FIELD_STATUS["__DEFAULT"] = "INITIAL";
    APPLICATION_FORM_FIELD_STATUS["INITIAL"] = "INITIAL";
    APPLICATION_FORM_FIELD_STATUS["STEP1"] = "STEP1";
    APPLICATION_FORM_FIELD_STATUS["STEP2"] = "STEP2";
    APPLICATION_FORM_FIELD_STATUS["STEP3"] = "STEP3";
})(APPLICATION_FORM_FIELD_STATUS = exports.APPLICATION_FORM_FIELD_STATUS || (exports.APPLICATION_FORM_FIELD_STATUS = {}));
exports.ALL_APPLICATION_FORM_FIELD_STATUS = common_1.getAllEnumStatuses(APPLICATION_FORM_FIELD_STATUS);
var FORM_FIELD_TYPES;
(function (FORM_FIELD_TYPES) {
    FORM_FIELD_TYPES["TEXT"] = "TEXT";
    FORM_FIELD_TYPES["MULTILINE_TEXT"] = "MULTILINE_TEXT";
    FORM_FIELD_TYPES["SELECT"] = "SELECT";
    FORM_FIELD_TYPES["SELECT_WITH_OTHER"] = "SELECT_WITH_OTHER";
    FORM_FIELD_TYPES["SELECT_WITH_RANGES"] = "SELECT_WITH_RANGES";
    FORM_FIELD_TYPES["PHONE"] = "PHONE";
    FORM_FIELD_TYPES["EMAIL"] = "EMAIL";
    FORM_FIELD_TYPES["DATE"] = "DATE";
    FORM_FIELD_TYPES["DATE_RANGE"] = "DATE_RANGE";
    FORM_FIELD_TYPES["NUMBER"] = "NUMBER";
    FORM_FIELD_TYPES["NUMBER_CURRENCY"] = "NUMBER_CURRENCY";
    FORM_FIELD_TYPES["BOOLEAN"] = "BOOLEAN";
    FORM_FIELD_TYPES["BOOLEAN_WITH_YES_ANSWER"] = "BOOLEAN_WITH_YES_ANSWER";
    FORM_FIELD_TYPES["BOOLEAN_WITH_NO_ANSWER"] = "BOOLEAN_WITH_NO_ANSWER";
    FORM_FIELD_TYPES["ARRAY_WITH_SUBFIELDS"] = "ARRAY_WITH_SUBFIELDS";
    FORM_FIELD_TYPES["PROFILE_PICTURE"] = "PROFILE_PICTURE";
    FORM_FIELD_TYPES["UPLOAD"] = "UPLOAD";
    FORM_FIELD_TYPES["CHECKBOX"] = "CHECKBOX";
    FORM_FIELD_TYPES["TOGGLE"] = "TOGGLE";
    FORM_FIELD_TYPES["SELECT_INSERT_TEXT"] = "SELECT_INSERT_TEXT";
    FORM_FIELD_TYPES["RENT_INCREASE_SCHEDULE"] = "RENT_INCREASE_SCHEDULE";
    FORM_FIELD_TYPES["RESIDENTS"] = "RESIDENTS";
    FORM_FIELD_TYPES["SIGNATURE"] = "SIGNATURE";
    FORM_FIELD_TYPES["SIGNATURE_TEXT"] = "SIGNATURE_TEXT";
    FORM_FIELD_TYPES["CONFIRMATION_TEXT"] = "CONFIRMATION_TEXT";
})(FORM_FIELD_TYPES = exports.FORM_FIELD_TYPES || (exports.FORM_FIELD_TYPES = {}));
exports.ALL_FORM_FIELD_TYPES = common_1.getAllEnumStatuses(FORM_FIELD_TYPES);
var SELECT_WITH_OTHER_VALUES;
(function (SELECT_WITH_OTHER_VALUES) {
    SELECT_WITH_OTHER_VALUES["OTHER"] = "OTHER";
})(SELECT_WITH_OTHER_VALUES = exports.SELECT_WITH_OTHER_VALUES || (exports.SELECT_WITH_OTHER_VALUES = {}));
var FEEDBACK_TYPE;
(function (FEEDBACK_TYPE) {
    FEEDBACK_TYPE["YOUR_WEBSITE"] = "YOUR_WEBSITE";
    FEEDBACK_TYPE["IMMOSCOUT24"] = "IMMOSCOUT24";
    FEEDBACK_TYPE["AD_IN_NEWSPAPER"] = "AD_IN_NEWSPAPER";
    FEEDBACK_TYPE["FRIENDS"] = "FRIENDS";
    FEEDBACK_TYPE["TENANT_RECOMMANDATION"] = "TENANT_RECOMMANDATION";
    FEEDBACK_TYPE["SEARCH_ENGINE"] = "SEARCH_ENGINE";
    FEEDBACK_TYPE["FACEBOOK"] = "FACEBOOK";
    FEEDBACK_TYPE["OTHER"] = "OTHER";
})(FEEDBACK_TYPE = exports.FEEDBACK_TYPE || (exports.FEEDBACK_TYPE = {}));
exports.ALL_FEEDBACK_TYPE = common_1.getAllEnumStatuses(FEEDBACK_TYPE);
var DEPOSIT_PAYMENT_TYPE;
(function (DEPOSIT_PAYMENT_TYPE) {
    DEPOSIT_PAYMENT_TYPE["KAUTIONSKASSE"] = "KAUTIONSKASSE";
    DEPOSIT_PAYMENT_TYPE["CASH"] = "CASH";
    DEPOSIT_PAYMENT_TYPE["SAVINGS_ACCOUNT"] = "SAVINGS_ACCOUNT";
    DEPOSIT_PAYMENT_TYPE["BANK_GUARANTEE"] = "BANK_GUARANTEE";
    DEPOSIT_PAYMENT_TYPE["OTHER"] = "OTHER";
})(DEPOSIT_PAYMENT_TYPE = exports.DEPOSIT_PAYMENT_TYPE || (exports.DEPOSIT_PAYMENT_TYPE = {}));
exports.ALL_DEPOSIT_PAYMENT_TYPE = common_1.getAllEnumStatuses(DEPOSIT_PAYMENT_TYPE);
