// @everreal/er-common-helpers/lib/helpers/constants/candidates.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var common_1 = require("./common");
var CANDIDATE_SCHEDULED_STATUS;
(function (CANDIDATE_SCHEDULED_STATUS) {
    CANDIDATE_SCHEDULED_STATUS["__DEFAULT"] = "NOT_INVITED";
    CANDIDATE_SCHEDULED_STATUS["NOT_INVITED"] = "NOT_INVITED";
    CANDIDATE_SCHEDULED_STATUS["INVITED"] = "INVITED";
    CANDIDATE_SCHEDULED_STATUS["ADMIN_CANCELLED"] = "ADMIN_CANCELLED";
    CANDIDATE_SCHEDULED_STATUS["CANDIDATE_CANCELLED"] = "CANDIDATE_CANCELLED";
    CANDIDATE_SCHEDULED_STATUS["SCHEDULED"] = "SCHEDULED";
    CANDIDATE_SCHEDULED_STATUS["NEW_TIMESLOTS_REQUESTED"] = "NEW_TIMESLOTS_REQUESTED";
})(CANDIDATE_SCHEDULED_STATUS = exports.CANDIDATE_SCHEDULED_STATUS || (exports.CANDIDATE_SCHEDULED_STATUS = {}));
exports.ALL_CANDIDATE_SCHEDULED_STATUS = common_1.getAllEnumStatuses(CANDIDATE_SCHEDULED_STATUS);
var CANDIDATE_VETTING_STATUS;
(function (CANDIDATE_VETTING_STATUS) {
    CANDIDATE_VETTING_STATUS["__DEFAULT"] = "NONE";
    CANDIDATE_VETTING_STATUS["NONE"] = "NONE";
    CANDIDATE_VETTING_STATUS["WAITING_FOR_INFORMATION"] = "WAITING_FOR_INFORMATION";
    CANDIDATE_VETTING_STATUS["SUBMITTED_INFORMATION"] = "SUBMITTED_INFORMATION";
})(CANDIDATE_VETTING_STATUS = exports.CANDIDATE_VETTING_STATUS || (exports.CANDIDATE_VETTING_STATUS = {}));
exports.ALL_CANDIDATE_VETTING_STATUS = common_1.getAllEnumStatuses(CANDIDATE_VETTING_STATUS);
var CANDIDATE_SOURCE;
(function (CANDIDATE_SOURCE) {
    CANDIDATE_SOURCE["__DEFAULT"] = "APPLIED_EVERREAL";
    CANDIDATE_SOURCE["APPLIED_EVERREAL"] = "APPLIED_EVERREAL";
    CANDIDATE_SOURCE["MANUAL_EVERREAL_CANDIDATE"] = "MANUAL_EVERREAL_CANDIDATE";
    CANDIDATE_SOURCE["MANUAL_EVERREAL_SCHEDULED"] = "MANUAL_EVERREAL_SCHEDULED";
    CANDIDATE_SOURCE["MANUAL_EVERREAL_APPLICANT"] = "MANUAL_EVERREAL_APPLICANT";
    CANDIDATE_SOURCE["IMMOSCOUT24"] = "IMMOSCOUT24";
    CANDIDATE_SOURCE["IMMOWELT"] = "IMMOWELT";
    CANDIDATE_SOURCE["IMMONET"] = "IMMONET";
    CANDIDATE_SOURCE["IVD24"] = "IVD24";
    CANDIDATE_SOURCE["NWZ"] = "NWZ";
    CANDIDATE_SOURCE["OPENIMMO_GENERIC"] = "OPENIMMO_GENERIC";
    CANDIDATE_SOURCE["EBAY"] = "EBAY";
    CANDIDATE_SOURCE["OFFLINE_CONTRACT"] = "OFFLINE_CONTRACT";
})(CANDIDATE_SOURCE = exports.CANDIDATE_SOURCE || (exports.CANDIDATE_SOURCE = {}));
exports.ALL_CANDIDATE_SOURCE = common_1.getAllEnumStatuses(CANDIDATE_SOURCE);
var CANDIDATE_OPTEDOUT_REASON;
(function (CANDIDATE_OPTEDOUT_REASON) {
    CANDIDATE_OPTEDOUT_REASON["MOVE_IN_DATE_LATE"] = "MOVE_IN_DATE_LATE";
    CANDIDATE_OPTEDOUT_REASON["PRICE_TOO_HIGH"] = "PRICE_TOO_HIGH";
    CANDIDATE_OPTEDOUT_REASON["FOUND_ANOTHER_APARTMENT"] = "FOUND_ANOTHER_APARTMENT";
    CANDIDATE_OPTEDOUT_REASON["LATE_RESPONSE_FROM_ADMIN"] = "LATE_RESPONSE_FROM_ADMIN";
    CANDIDATE_OPTEDOUT_REASON["AMENITIES"] = "AMENITIES";
    CANDIDATE_OPTEDOUT_REASON["LAYOUT"] = "LAYOUT";
    CANDIDATE_OPTEDOUT_REASON["LOCATION"] = "LOCATION";
    CANDIDATE_OPTEDOUT_REASON["EXPENSIVE_PARKING_SPOT"] = "EXPENSIVE_PARKING_SPOT";
    CANDIDATE_OPTEDOUT_REASON["OTHER"] = "OTHER";
})(CANDIDATE_OPTEDOUT_REASON = exports.CANDIDATE_OPTEDOUT_REASON || (exports.CANDIDATE_OPTEDOUT_REASON = {}));
exports.ALL_CANDIDATE_OPTEDOUT_REASON = common_1.getAllEnumStatuses(CANDIDATE_OPTEDOUT_REASON);
var UNSUBSCRIBED_BY_ADMIN_REASON;
(function (UNSUBSCRIBED_BY_ADMIN_REASON) {
    UNSUBSCRIBED_BY_ADMIN_REASON["MOVE_IN_DATE_LATE"] = "MOVE_IN_DATE_LATE";
    UNSUBSCRIBED_BY_ADMIN_REASON["PRICE_TOO_HIGH"] = "PRICE_TOO_HIGH";
    UNSUBSCRIBED_BY_ADMIN_REASON["FOUND_ANOTHER_APARTMENT"] = "FOUND_ANOTHER_APARTMENT";
    UNSUBSCRIBED_BY_ADMIN_REASON["AMENITIES"] = "AMENITIES";
    UNSUBSCRIBED_BY_ADMIN_REASON["LAYOUT"] = "LAYOUT";
    UNSUBSCRIBED_BY_ADMIN_REASON["LOCATION"] = "LOCATION";
    UNSUBSCRIBED_BY_ADMIN_REASON["EXPENSIVE_PARKING_SPOT"] = "EXPENSIVE_PARKING_SPOT";
    UNSUBSCRIBED_BY_ADMIN_REASON["OTHER"] = "OTHER";
})(UNSUBSCRIBED_BY_ADMIN_REASON = exports.UNSUBSCRIBED_BY_ADMIN_REASON || (exports.UNSUBSCRIBED_BY_ADMIN_REASON = {}));
exports.ALL_UNSUBSCRIBED_BY_ADMIN_REASON = common_1.getAllEnumStatuses(UNSUBSCRIBED_BY_ADMIN_REASON);
var CANDIDATE_DISABLED_STATUS;
(function (CANDIDATE_DISABLED_STATUS) {
    CANDIDATE_DISABLED_STATUS["__DEFAULT"] = "NONE";
    CANDIDATE_DISABLED_STATUS["NONE"] = "NONE";
    CANDIDATE_DISABLED_STATUS["REJECTED_BY_ADMIN"] = "REJECTED_BY_ADMIN";
    CANDIDATE_DISABLED_STATUS["UNSUBSCRIBED_BY_ADMIN"] = "UNSUBSCRIBED_BY_ADMIN";
    CANDIDATE_DISABLED_STATUS["CANDIDATE_OPTED_OUT"] = "CANDIDATE_OPTED_OUT";
})(CANDIDATE_DISABLED_STATUS = exports.CANDIDATE_DISABLED_STATUS || (exports.CANDIDATE_DISABLED_STATUS = {}));
exports.ALL_CANDIDATE_DISABLED_STATUS = common_1.getAllEnumStatuses(CANDIDATE_DISABLED_STATUS);
