// @everreal/er-common-helpers/lib/helpers/constants/common.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAllStatuses = function (value) {
    return Object.keys(value)
        .filter(function (key) { return key !== '__DEFAULT'; })
        .map(function (key) { return value[key]; });
};
exports.getAllEnumStatuses = function (value) {
    return Object.keys(value)
        .filter(function (key) { return key !== '__DEFAULT'; })
        .map(function (key) { return value[key]; });
};
exports.ALL_GENDER = ['MALE', 'FEMALE', 'NO_INFORMATION'];
var YES_NO_DEPENDS;
(function (YES_NO_DEPENDS) {
    YES_NO_DEPENDS["YES"] = "YES";
    YES_NO_DEPENDS["NO"] = "NO";
    YES_NO_DEPENDS["DEPENDS"] = "DEPENDS";
})(YES_NO_DEPENDS = exports.YES_NO_DEPENDS || (exports.YES_NO_DEPENDS = {}));
exports.ALL_YES_NO_DEPENDS = exports.getAllEnumStatuses(YES_NO_DEPENDS);
var CURRENCY_TYPE;
(function (CURRENCY_TYPE) {
    CURRENCY_TYPE["__DEFAULT"] = "EUR";
    CURRENCY_TYPE["EUR"] = "EUR";
    CURRENCY_TYPE["USD"] = "USD";
})(CURRENCY_TYPE = exports.CURRENCY_TYPE || (exports.CURRENCY_TYPE = {}));
exports.ALL_CURRENCY_TYPE = exports.getAllEnumStatuses(CURRENCY_TYPE);
var TIME_INTERVALS;
(function (TIME_INTERVALS) {
    TIME_INTERVALS[TIME_INTERVALS["ONE_MINUTE"] = 60000] = "ONE_MINUTE";
    TIME_INTERVALS[TIME_INTERVALS["FIVE_MINUTES"] = 300000] = "FIVE_MINUTES";
    TIME_INTERVALS[TIME_INTERVALS["ONE_HOUR"] = 3600000] = "ONE_HOUR";
    TIME_INTERVALS[TIME_INTERVALS["ONE_DAY"] = 86400000] = "ONE_DAY";
    TIME_INTERVALS[TIME_INTERVALS["ONE_WEEK"] = 604800000] = "ONE_WEEK";
    TIME_INTERVALS[TIME_INTERVALS["ONE_MONTH"] = 2592000000] = "ONE_MONTH";
    TIME_INTERVALS[TIME_INTERVALS["ONE_YEAR"] = 31536000000] = "ONE_YEAR";
})(TIME_INTERVALS = exports.TIME_INTERVALS || (exports.TIME_INTERVALS = {}));
var COMPARISON_SIGNS;
(function (COMPARISON_SIGNS) {
    COMPARISON_SIGNS["__DEFAULT"] = "EQUALS";
    COMPARISON_SIGNS["EQUALS"] = "EQUALS";
    COMPARISON_SIGNS["GREATER_THAN"] = "GREATER_THAN";
    COMPARISON_SIGNS["GREATER_THAN_EQUAL"] = "GREATER_THAN_EQUAL";
    COMPARISON_SIGNS["LOWER_THAN"] = "LOWER_THAN";
    COMPARISON_SIGNS["LOWER_THAN_EQUAL"] = "LOWER_THAN_EQUAL";
    COMPARISON_SIGNS["STRING_CONTAINS"] = "STRING_CONTAINS";
})(COMPARISON_SIGNS = exports.COMPARISON_SIGNS || (exports.COMPARISON_SIGNS = {}));
exports.ALL_COMPARISON_SIGNS = exports.getAllEnumStatuses(COMPARISON_SIGNS);
var MATH_OPERATORS;
(function (MATH_OPERATORS) {
    MATH_OPERATORS["PLUS"] = "PLUS";
    MATH_OPERATORS["MINUS"] = "MINUS";
    MATH_OPERATORS["DIVIDE"] = "DIVIDE";
    MATH_OPERATORS["MULTIPLY"] = "MULTIPLY";
    MATH_OPERATORS["NONE"] = "NONE";
})(MATH_OPERATORS = exports.MATH_OPERATORS || (exports.MATH_OPERATORS = {}));
exports.ALL_MATH_OPERATORS = exports.getAllEnumStatuses(MATH_OPERATORS);
var REASON_FOR_MOVING;
(function (REASON_FOR_MOVING) {
    REASON_FOR_MOVING["HOUSING_SIZE"] = "HOUSING_SIZE";
    REASON_FOR_MOVING["NEW_JOB"] = "NEW_JOB";
    REASON_FOR_MOVING["SAVING_ON_RENT"] = "SAVING_ON_RENT";
    REASON_FOR_MOVING["NEW_BUSINESS_FOUNDER"] = "NEW_BUSINESS_FOUNDER";
    REASON_FOR_MOVING["RELOCATION"] = "RELOCATION";
    REASON_FOR_MOVING["OTHER"] = "OTHER";
})(REASON_FOR_MOVING = exports.REASON_FOR_MOVING || (exports.REASON_FOR_MOVING = {}));
exports.ALL_REASON_FOR_MOVING = exports.getAllEnumStatuses(REASON_FOR_MOVING);
var EMPLOYMENT_TYPE;
(function (EMPLOYMENT_TYPE) {
    EMPLOYMENT_TYPE["EMPLOYED"] = "EMPLOYED";
    EMPLOYMENT_TYPE["SELF_EMPLOYED"] = "SELF_EMPLOYED";
    EMPLOYMENT_TYPE["STUDENT"] = "STUDENT";
    EMPLOYMENT_TYPE["SEEKING_WORK"] = "SEEKING_WORK";
    EMPLOYMENT_TYPE["CLERK"] = "CLERK";
    EMPLOYMENT_TYPE["RETIRED"] = "RETIRED";
    EMPLOYMENT_TYPE["HOUSE_MAN_WIFE"] = "HOUSE_MAN_WIFE";
    EMPLOYMENT_TYPE["APPRENTICE"] = "APPRENTICE";
    EMPLOYMENT_TYPE["POSTGRADUATE"] = "POSTGRADUATE";
    EMPLOYMENT_TYPE["OTHER"] = "OTHER";
})(EMPLOYMENT_TYPE = exports.EMPLOYMENT_TYPE || (exports.EMPLOYMENT_TYPE = {}));
exports.ALL_EMPLOYMENT_TYPE = exports.getAllEnumStatuses(EMPLOYMENT_TYPE);
var EMPLOYMENT_CONTRACT_TYPE;
(function (EMPLOYMENT_CONTRACT_TYPE) {
    EMPLOYMENT_CONTRACT_TYPE["LIMITED"] = "LIMITED";
    EMPLOYMENT_CONTRACT_TYPE["UNLIMITED"] = "UNLIMITED";
    EMPLOYMENT_CONTRACT_TYPE["PROBATION"] = "PROBATION";
    EMPLOYMENT_CONTRACT_TYPE["OTHER"] = "OTHER";
})(EMPLOYMENT_CONTRACT_TYPE = exports.EMPLOYMENT_CONTRACT_TYPE || (exports.EMPLOYMENT_CONTRACT_TYPE = {}));
exports.ALL_EMPLOYMENT_CONTRACT_TYPE = exports.getAllEnumStatuses(EMPLOYMENT_CONTRACT_TYPE);
var INCOME_RANGES;
(function (INCOME_RANGES) {
    INCOME_RANGES["RANGE_0_500"] = "RANGE_0_500";
    INCOME_RANGES["RANGE_500_1000"] = "RANGE_500_1000";
    INCOME_RANGES["RANGE_1000_1500"] = "RANGE_1000_1500";
    INCOME_RANGES["RANGE_1500_2000"] = "RANGE_1500_2000";
    INCOME_RANGES["RANGE_2000_2500"] = "RANGE_2000_2500";
    INCOME_RANGES["RANGE_2500_3000"] = "RANGE_2500_3000";
    INCOME_RANGES["RANGE_3000_3500"] = "RANGE_3000_3500";
    INCOME_RANGES["RANGE_3500_4000"] = "RANGE_3500_4000";
    INCOME_RANGES["RANGE_4000_4500"] = "RANGE_4000_4500";
    INCOME_RANGES["RANGE_4500_5000"] = "RANGE_4500_5000";
    INCOME_RANGES["RANGE_5000_5500"] = "RANGE_5000_5500";
    INCOME_RANGES["RANGE_6000_6500"] = "RANGE_6000_6500";
    INCOME_RANGES["RANGE_6500_7000"] = "RANGE_6500_7000";
    INCOME_RANGES["GREATER_7000"] = "GREATER_7000";
})(INCOME_RANGES = exports.INCOME_RANGES || (exports.INCOME_RANGES = {}));
exports.ALL_INCOME_RANGES = exports.getAllEnumStatuses(INCOME_RANGES);
var REASON_FOR_NEW_SPACE;
(function (REASON_FOR_NEW_SPACE) {
    REASON_FOR_NEW_SPACE["NEW_BUSINESS_FOUNDATION"] = "NEW_BUSINESS_FOUNDATION";
    REASON_FOR_NEW_SPACE["BUSINESS_EXPANSION"] = "BUSINESS_EXPANSION";
    REASON_FOR_NEW_SPACE["NEW_SUBSIDIARY"] = "NEW_SUBSIDIARY";
    REASON_FOR_NEW_SPACE["REDUCING_COST"] = "REDUCING_COST";
    REASON_FOR_NEW_SPACE["BETTER_AMENITIES"] = "BETTER_AMENITIES";
    REASON_FOR_NEW_SPACE["BETTER_LOCATION"] = "BETTER_LOCATION";
})(REASON_FOR_NEW_SPACE = exports.REASON_FOR_NEW_SPACE || (exports.REASON_FOR_NEW_SPACE = {}));
exports.ALL_REASON_FOR_NEW_SPACE = exports.getAllEnumStatuses(REASON_FOR_NEW_SPACE);
