// @everreal/er-common-helpers/lib/helpers/constants/contract.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var common_1 = require("./common");
var CONTRACT_STATUS;
(function (CONTRACT_STATUS) {
    CONTRACT_STATUS["__DEFAULT"] = "NOT_INITIALISED";
    CONTRACT_STATUS["WAIT_FOR_TENANT"] = "WAIT_FOR_TENANT";
    CONTRACT_STATUS["DOWNLOADED_TENANT"] = "DOWNLOADED_TENANT";
    CONTRACT_STATUS["SUBMITTED_TENANT"] = "SUBMITTED_TENANT";
    CONTRACT_STATUS["DOWNLOADED_ADMIN"] = "DOWNLOADED_ADMIN";
    CONTRACT_STATUS["SUBMITTED_ADMIN"] = "SUBMITTED_ADMIN";
    CONTRACT_STATUS["OFFLINE_CONTRACT"] = "OFFLINE_CONTRACT";
    CONTRACT_STATUS["CONTRACT_TERMINATED"] = "CONTRACT_TERMINATED";
    CONTRACT_STATUS["CONTRACT_UPDATED"] = "CONTRACT_UPDATED";
    CONTRACT_STATUS["CONTRACT_TERMINATION_REVERTED"] = "CONTRACT_TERMINATION_REVERTED";
})(CONTRACT_STATUS = exports.CONTRACT_STATUS || (exports.CONTRACT_STATUS = {}));
exports.ALL_CONTRACT_STATUS = common_1.getAllEnumStatuses(CONTRACT_STATUS);
exports.APARTMENT_RENOVATED_STATUS = {
    IS_RENOVATED: 'IS_RENOVATED',
    IS_NOT_RENOVATED: 'IS_NOT_RENOVATED',
};
exports.ALL_APARTMENT_RENOVATED_STATUS = common_1.getAllStatuses(exports.APARTMENT_RENOVATED_STATUS);
exports.TERMINATION_REASON = {
    APARTMENT_TOO_BIG: 'APARTMENT_TOO_BIG',
    APARTMENT_TOO_SMALL: 'APARTMENT_TOO_SMALL',
    UTILITY_COST_TOO_HIGH: 'UTILITY_COST_TOO_HIGH',
    RENT_TOO_HIGH: 'RENT_TOO_HIGH',
    LOCATION: 'LOCATION',
    NEIGHBORHOOD: 'NEIGHBORHOOD',
    JOB_CHANGE: 'JOB_CHANGE',
    MOVING_TO_OWN_HOME: 'MOVING_TO_OWN_HOME',
    SENIOR_HOME: 'SENIOR_HOME',
    DECEASED: 'DECEASED',
    DEFECTIVE_APARTMENT: 'DEFECTIVE_APARTMENT',
    DEFECTIVE_PROPERTY: 'DEFECTIVE_PROPERTY',
    SERVICE_CARETAKER: 'SERVICE_CARETAKER',
    SERVICE_ADMINISTRATOR: 'SERVICE_ADMINISTRATOR',
    TERMINATION_BY_OWNER: 'TERMINATION_BY_OWNER',
    OTHER: 'OTHER',
};
exports.ALL_TERMINATION_REASON = common_1.getAllStatuses(exports.TERMINATION_REASON);
