// @everreal/er-common-helpers/lib/helpers/constants/integrations.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var common_1 = require("./common");
var INTEGRATION_CATEGORY;
(function (INTEGRATION_CATEGORY) {
    INTEGRATION_CATEGORY["OAUTH1"] = "OAUTH1";
    INTEGRATION_CATEGORY["OPENIMMO_XML_FTP"] = "OPENIMMO_XML_FTP";
    INTEGRATION_CATEGORY["BANK_ACCOUNTS"] = "BANK_ACCOUNTS";
    INTEGRATION_CATEGORY["CREDIT_CHECK"] = "CREDIT_CHECK";
})(INTEGRATION_CATEGORY = exports.INTEGRATION_CATEGORY || (exports.INTEGRATION_CATEGORY = {}));
exports.ALL_INTEGRATION_CATEGORY = common_1.getAllEnumStatuses(INTEGRATION_CATEGORY);
var INTEGRATION_TYPE;
(function (INTEGRATION_TYPE) {
    INTEGRATION_TYPE["NONE"] = "NONE";
    INTEGRATION_TYPE["IMMOSCOUT24"] = "IMMOSCOUT24";
    INTEGRATION_TYPE["IMMOWELT"] = "IMMOWELT";
    INTEGRATION_TYPE["IMMONET"] = "IMMONET";
    INTEGRATION_TYPE["EBAY"] = "EBAY";
    INTEGRATION_TYPE["OPENIMMO_GENERIC"] = "OPENIMMO_GENERIC";
    INTEGRATION_TYPE["NWZ"] = "NWZ";
    INTEGRATION_TYPE["IVD24"] = "IVD24";
    INTEGRATION_TYPE["HAUSBANK"] = "HAUSBANK";
    INTEGRATION_TYPE["SCHUFA"] = "SCHUFA";
})(INTEGRATION_TYPE = exports.INTEGRATION_TYPE || (exports.INTEGRATION_TYPE = {}));
exports.ALL_INTEGRATION_TYPE = common_1.getAllEnumStatuses(INTEGRATION_TYPE);
var INTEGRATION_CATEGORIES;
(function (INTEGRATION_CATEGORIES) {
    INTEGRATION_CATEGORIES["IMMOSCOUT24"] = "OAUTH1";
    INTEGRATION_CATEGORIES["IMMOWELT"] = "OPENIMMO_XML_FTP";
    INTEGRATION_CATEGORIES["IMMONET"] = "OPENIMMO_XML_FTP";
    INTEGRATION_CATEGORIES["EBAY"] = "OPENIMMO_XML_FTP";
    INTEGRATION_CATEGORIES["OPENIMMO_GENERIC"] = "OPENIMMO_XML_FTP";
    INTEGRATION_CATEGORIES["NWZ"] = "OPENIMMO_XML_FTP";
    INTEGRATION_CATEGORIES["IVD24"] = "OPENIMMO_XML_FTP";
    INTEGRATION_CATEGORIES["HAUSBANK"] = "BANK_ACCOUNTS";
    INTEGRATION_CATEGORIES["SCHUFA"] = "CREDIT_CHECK";
})(INTEGRATION_CATEGORIES = exports.INTEGRATION_CATEGORIES || (exports.INTEGRATION_CATEGORIES = {}));
var ADVERTISE_STATUS;
(function (ADVERTISE_STATUS) {
    ADVERTISE_STATUS["CREATED"] = "CREATED";
    ADVERTISE_STATUS["PUBLISHED"] = "PUBLISHED";
    ADVERTISE_STATUS["PENDING_PUBLISHING"] = "PENDING_PUBLISHING";
    ADVERTISE_STATUS["UNPUBLISHED"] = "UNPUBLISHED";
    ADVERTISE_STATUS["PENDING_UNPUBLISHING"] = "PENDING_UNPUBLISHING";
})(ADVERTISE_STATUS = exports.ADVERTISE_STATUS || (exports.ADVERTISE_STATUS = {}));
exports.ALL_ADVERTISE_STATUS = common_1.getAllEnumStatuses(ADVERTISE_STATUS);
