// @everreal/er-common-helpers/lib/helpers/constants/listing.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var common_1 = require("./common");
var LISTING_STATUS;
(function (LISTING_STATUS) {
    LISTING_STATUS["__DEFAULT"] = "OPEN_FOR_CANDIDATES";
    LISTING_STATUS["OPEN_FOR_CANDIDATES"] = "OPEN_FOR_CANDIDATES";
    LISTING_STATUS["OPEN_FOR_APPLICANTS"] = "OPEN_FOR_APPLICANTS";
    LISTING_STATUS["CONTRACTING_STARTED"] = "CONTRACTING_STARTED";
    LISTING_STATUS["CONTRACTING_COMPLETED"] = "CONTRACTING_COMPLETED";
    LISTING_STATUS["MOVE_IN_COMPLETED"] = "MOVE_IN_COMPLETED";
    LISTING_STATUS["IS_INACTIVE"] = "IS_INACTIVE";
    LISTING_STATUS["IS_ARCHIVED"] = "IS_ARCHIVED";
})(LISTING_STATUS = exports.LISTING_STATUS || (exports.LISTING_STATUS = {}));
exports.ALL_LISTING_STATUS = common_1.getAllEnumStatuses(LISTING_STATUS);
/**
 * https://basarat.gitbooks.io/typescript/docs/enums.html#enums-as-flags
 */
var LISTING_STATUS_FLAGS;
(function (LISTING_STATUS_FLAGS) {
    LISTING_STATUS_FLAGS[LISTING_STATUS_FLAGS["NONE"] = 0] = "NONE";
    LISTING_STATUS_FLAGS[LISTING_STATUS_FLAGS["OPEN_FOR_CANDIDATES"] = 1] = "OPEN_FOR_CANDIDATES";
    LISTING_STATUS_FLAGS[LISTING_STATUS_FLAGS["OPEN_FOR_APPLICANTS"] = 2] = "OPEN_FOR_APPLICANTS";
    LISTING_STATUS_FLAGS[LISTING_STATUS_FLAGS["CONTRACTING_STARTED"] = 4] = "CONTRACTING_STARTED";
    LISTING_STATUS_FLAGS[LISTING_STATUS_FLAGS["CONTRACTING_COMPLETED"] = 8] = "CONTRACTING_COMPLETED";
    LISTING_STATUS_FLAGS[LISTING_STATUS_FLAGS["MOVE_IN_COMPLETED"] = 16] = "MOVE_IN_COMPLETED";
})(LISTING_STATUS_FLAGS = exports.LISTING_STATUS_FLAGS || (exports.LISTING_STATUS_FLAGS = {}));
var LISTING_MANAGED_BY;
(function (LISTING_MANAGED_BY) {
    LISTING_MANAGED_BY["OWNER"] = "OWNER";
    LISTING_MANAGED_BY["MANAGEMENT_COMPANY"] = "MANAGEMENT_COMPANY";
})(LISTING_MANAGED_BY = exports.LISTING_MANAGED_BY || (exports.LISTING_MANAGED_BY = {}));
exports.ALL_LISTING_MANAGED_BY = common_1.getAllEnumStatuses(LISTING_MANAGED_BY);
var AMENITIES_CONDITION;
(function (AMENITIES_CONDITION) {
    AMENITIES_CONDITION["FIRST_TIME_USE"] = "FIRST_TIME_USE";
    AMENITIES_CONDITION["FIRST_TIME_USE_AFTER_REFURBISHMENT"] = "FIRST_TIME_USE_AFTER_REFURBISHMENT";
    AMENITIES_CONDITION["AS_GOOD_AS_NEW"] = "AS_GOOD_AS_NEW";
    AMENITIES_CONDITION["REFURBISHED"] = "REFURBISHED";
    AMENITIES_CONDITION["UPGRADED"] = "UPGRADED";
    AMENITIES_CONDITION["FULLY_RENOVATED"] = "FULLY_RENOVATED";
    AMENITIES_CONDITION["WELL_KEPT"] = "WELL_KEPT";
    AMENITIES_CONDITION["IN_NEED_OF_RENOVATION"] = "IN_NEED_OF_RENOVATION";
    AMENITIES_CONDITION["NEGOTIABLE"] = "NEGOTIABLE";
    AMENITIES_CONDITION["DILAPIDATED"] = "DILAPIDATED";
})(AMENITIES_CONDITION = exports.AMENITIES_CONDITION || (exports.AMENITIES_CONDITION = {}));
exports.ALL_AMENITIES_CONDITION = common_1.getAllEnumStatuses(AMENITIES_CONDITION);
var AMENITIES_INCLUDED;
(function (AMENITIES_INCLUDED) {
    AMENITIES_INCLUDED["BUILTIN_KITCHEN"] = "BUILTIN_KITCHEN";
    AMENITIES_INCLUDED["ELEVATOR"] = "ELEVATOR";
    AMENITIES_INCLUDED["GOODS_LIFT"] = "GOODS_LIFT";
    AMENITIES_INCLUDED["BALCONY_OR_TERRACE"] = "BALCONY_OR_TERRACE";
    AMENITIES_INCLUDED["GUEST_TOILET"] = "GUEST_TOILET";
    AMENITIES_INCLUDED["GARDEN"] = "GARDEN";
    AMENITIES_INCLUDED["BASEMENT"] = "BASEMENT";
    AMENITIES_INCLUDED["STEPLESS_ENTRY"] = "STEPLESS_ENTRY";
    AMENITIES_INCLUDED["LIVING_QUALIFICATION_CERTIFICATE"] = "LIVING_QUALIFICATION_CERTIFICATE";
    AMENITIES_INCLUDED["SUITABLE_FOR_SHARED_APARTMENT"] = "SUITABLE_FOR_SHARED_APARTMENT";
    AMENITIES_INCLUDED["SUITABLE_FOR_HOLIDAY"] = "SUITABLE_FOR_HOLIDAY";
    AMENITIES_INCLUDED["CAFETERIA"] = "CAFETERIA";
    AMENITIES_INCLUDED["HIGH_VOLTAGE"] = "HIGH_VOLTAGE";
    AMENITIES_INCLUDED["RAMP"] = "RAMP";
    AMENITIES_INCLUDED["HYDRAULIC_RAMP"] = "HYDRAULIC_RAMP";
    AMENITIES_INCLUDED["TERRACE"] = "TERRACE";
    AMENITIES_INCLUDED["HAS_CRANE_RUNWAY"] = "HAS_CRANE_RUNWAY";
})(AMENITIES_INCLUDED = exports.AMENITIES_INCLUDED || (exports.AMENITIES_INCLUDED = {}));
exports.ALL_AMENITIES_INCLUDED = common_1.getAllEnumStatuses(AMENITIES_INCLUDED);
var HEATING_TYPE;
(function (HEATING_TYPE) {
    HEATING_TYPE["BLOCK_HEATING_STATION"] = "BLOCK_HEATING_STATION";
    HEATING_TYPE["ELECTRIC_HEATING"] = "ELECTRIC_HEATING";
    HEATING_TYPE["SELF_CONTAINED_CENTRAL_HEATING"] = "SELF_CONTAINED_CENTRAL_HEATING";
    HEATING_TYPE["TELEHEATING"] = "TELEHEATING";
    HEATING_TYPE["FLOOR_HEATING"] = "FLOOR_HEATING";
    HEATING_TYPE["GAS_HEATING"] = "GAS_HEATING";
    HEATING_TYPE["WOOD_PELLET_HEATER"] = "WOOD_PELLET_HEATER";
    HEATING_TYPE["NIGHT_STORAGE_HEATER"] = "NIGHT_STORAGE_HEATER";
    HEATING_TYPE["STOVE_HEATING"] = "STOVE_HEATING";
    HEATING_TYPE["OIL_HEATING"] = "OIL_HEATING";
    HEATING_TYPE["SOLAR_HEATING"] = "SOLAR_HEATING";
    HEATING_TYPE["HEAT_PUMP"] = "HEAT_PUMP";
    HEATING_TYPE["CENTRAL_HEATING"] = "CENTRAL_HEATING";
})(HEATING_TYPE = exports.HEATING_TYPE || (exports.HEATING_TYPE = {}));
exports.ALL_HEATING_TYPE = common_1.getAllEnumStatuses(HEATING_TYPE);
var QUALITY_OF_AMENITIES;
(function (QUALITY_OF_AMENITIES) {
    QUALITY_OF_AMENITIES["LUXURIOUS"] = "LUXURIOUS";
    QUALITY_OF_AMENITIES["UPSCALE"] = "UPSCALE";
    QUALITY_OF_AMENITIES["NORMAL"] = "NORMAL";
    QUALITY_OF_AMENITIES["BASIC"] = "BASIC";
})(QUALITY_OF_AMENITIES = exports.QUALITY_OF_AMENITIES || (exports.QUALITY_OF_AMENITIES = {}));
exports.ALL_QUALITY_OF_AMENITIES = common_1.getAllEnumStatuses(QUALITY_OF_AMENITIES);
var PROPERTY_OWNERSHIP_TYPE;
(function (PROPERTY_OWNERSHIP_TYPE) {
    PROPERTY_OWNERSHIP_TYPE["INDIVIDUAL"] = "INDIVIDUAL";
    PROPERTY_OWNERSHIP_TYPE["MULTIPLE"] = "MULTIPLE";
})(PROPERTY_OWNERSHIP_TYPE = exports.PROPERTY_OWNERSHIP_TYPE || (exports.PROPERTY_OWNERSHIP_TYPE = {}));
exports.ALL_PROPERTY_OWNERSHIP_TYPE = common_1.getAllEnumStatuses(PROPERTY_OWNERSHIP_TYPE);
var EDIT_PROPERTY_STATUS;
(function (EDIT_PROPERTY_STATUS) {
    EDIT_PROPERTY_STATUS["CREATE"] = "CREATE";
    EDIT_PROPERTY_STATUS["SELECT"] = "SELECT";
    EDIT_PROPERTY_STATUS["NONE"] = "NONE";
})(EDIT_PROPERTY_STATUS = exports.EDIT_PROPERTY_STATUS || (exports.EDIT_PROPERTY_STATUS = {}));
exports.ALL_EDIT_PROPERTY_STATUS = common_1.getAllEnumStatuses(EDIT_PROPERTY_STATUS);
var EDIT_UNIT_STATUS;
(function (EDIT_UNIT_STATUS) {
    EDIT_UNIT_STATUS["CREATE"] = "CREATE";
    EDIT_UNIT_STATUS["SELECT"] = "SELECT";
    EDIT_UNIT_STATUS["NONE"] = "NONE";
})(EDIT_UNIT_STATUS = exports.EDIT_UNIT_STATUS || (exports.EDIT_UNIT_STATUS = {}));
exports.ALL_EDIT_UNIT_STATUS = common_1.getAllEnumStatuses(EDIT_UNIT_STATUS);
var YES_NO;
(function (YES_NO) {
    YES_NO["YES"] = "YES";
    YES_NO["NO"] = "NO";
})(YES_NO = exports.YES_NO || (exports.YES_NO = {}));
exports.ALL_YES_NO = common_1.getAllEnumStatuses(YES_NO);
var RENTAL_INTERVAL_TYPE_SHORT_TERM_APARTMENT;
(function (RENTAL_INTERVAL_TYPE_SHORT_TERM_APARTMENT) {
    RENTAL_INTERVAL_TYPE_SHORT_TERM_APARTMENT["DAY"] = "DAY";
    RENTAL_INTERVAL_TYPE_SHORT_TERM_APARTMENT["WEEK"] = "WEEK";
    RENTAL_INTERVAL_TYPE_SHORT_TERM_APARTMENT["MONTH"] = "MONTH";
})(RENTAL_INTERVAL_TYPE_SHORT_TERM_APARTMENT = exports.RENTAL_INTERVAL_TYPE_SHORT_TERM_APARTMENT || (exports.RENTAL_INTERVAL_TYPE_SHORT_TERM_APARTMENT = {}));
exports.ALL_RENTAL_INTERVAL_TYPE_SHORT_TERM_APARTMENT = common_1.getAllEnumStatuses(RENTAL_INTERVAL_TYPE_SHORT_TERM_APARTMENT);
var UPLOADING_STATUS;
(function (UPLOADING_STATUS) {
    UPLOADING_STATUS["NONE"] = "NONE";
    UPLOADING_STATUS["UPLOADING"] = "UPLOADING";
})(UPLOADING_STATUS = exports.UPLOADING_STATUS || (exports.UPLOADING_STATUS = {}));
exports.ALL_UPLOADING_STATUS = common_1.getAllEnumStatuses(UPLOADING_STATUS);
var REJECTION_MESSAGE_TYPE;
(function (REJECTION_MESSAGE_TYPE) {
    REJECTION_MESSAGE_TYPE["__DEFAULT"] = "STANDARD";
    REJECTION_MESSAGE_TYPE["STANDARD"] = "STANDARD";
    REJECTION_MESSAGE_TYPE["GLOBAL_CUSTOM"] = "GLOBAL_CUSTOM";
    REJECTION_MESSAGE_TYPE["CUSTOM"] = "CUSTOM";
})(REJECTION_MESSAGE_TYPE = exports.REJECTION_MESSAGE_TYPE || (exports.REJECTION_MESSAGE_TYPE = {}));
exports.ALL_REJECTION_MESSAGE_TYPE = common_1.getAllEnumStatuses(REJECTION_MESSAGE_TYPE);
var ENERGY_PERFORMANCE_CERTIFICATE_AVAILABILITY;
(function (ENERGY_PERFORMANCE_CERTIFICATE_AVAILABILITY) {
    ENERGY_PERFORMANCE_CERTIFICATE_AVAILABILITY["AVAILABLE"] = "AVAILABLE";
    ENERGY_PERFORMANCE_CERTIFICATE_AVAILABILITY["AVAILABLE_AT_VIEWING"] = "AVAILABLE_AT_VIEWING";
    ENERGY_PERFORMANCE_CERTIFICATE_AVAILABILITY["NOT_REQUIRED"] = "NOT_REQUIRED";
})(ENERGY_PERFORMANCE_CERTIFICATE_AVAILABILITY = exports.ENERGY_PERFORMANCE_CERTIFICATE_AVAILABILITY || (exports.ENERGY_PERFORMANCE_CERTIFICATE_AVAILABILITY = {}));
exports.ALL_ENERGY_PERFORMANCE_CERTIFICATE_AVAILABILITY = common_1.getAllEnumStatuses(ENERGY_PERFORMANCE_CERTIFICATE_AVAILABILITY);
var ENERGY_CERTIFICATE_CREATION_DATE;
(function (ENERGY_CERTIFICATE_CREATION_DATE) {
    ENERGY_CERTIFICATE_CREATION_DATE["BEFORE_01_MAY_2014"] = "BEFORE_01_MAY_2014";
    ENERGY_CERTIFICATE_CREATION_DATE["FROM_01_MAY_2014"] = "FROM_01_MAY_2014";
})(ENERGY_CERTIFICATE_CREATION_DATE = exports.ENERGY_CERTIFICATE_CREATION_DATE || (exports.ENERGY_CERTIFICATE_CREATION_DATE = {}));
exports.ALL_ENERGY_CERTIFICATE_CREATION_DATE = common_1.getAllEnumStatuses(ENERGY_CERTIFICATE_CREATION_DATE);
var BUILDING_ENERGY_RATING_TYPE;
(function (BUILDING_ENERGY_RATING_TYPE) {
    BUILDING_ENERGY_RATING_TYPE["ENERGY_REQUIRED"] = "ENERGY_REQUIRED";
    BUILDING_ENERGY_RATING_TYPE["ENERGY_CONSUMPTION"] = "ENERGY_CONSUMPTION";
})(BUILDING_ENERGY_RATING_TYPE = exports.BUILDING_ENERGY_RATING_TYPE || (exports.BUILDING_ENERGY_RATING_TYPE = {}));
exports.ALL_BUILDING_ENERGY_RATING_TYPE = common_1.getAllEnumStatuses(BUILDING_ENERGY_RATING_TYPE);
var ENERGY_EFFICIENCY_CLASS;
(function (ENERGY_EFFICIENCY_CLASS) {
    ENERGY_EFFICIENCY_CLASS["NO_INFORMATION"] = "NO_INFORMATION";
    ENERGY_EFFICIENCY_CLASS["A_PLUS"] = "A_PLUS";
    ENERGY_EFFICIENCY_CLASS["A"] = "A";
    ENERGY_EFFICIENCY_CLASS["B"] = "B";
    ENERGY_EFFICIENCY_CLASS["C"] = "C";
    ENERGY_EFFICIENCY_CLASS["D"] = "D";
    ENERGY_EFFICIENCY_CLASS["E"] = "E";
    ENERGY_EFFICIENCY_CLASS["F"] = "F";
    ENERGY_EFFICIENCY_CLASS["G"] = "G";
    ENERGY_EFFICIENCY_CLASS["H"] = "H";
})(ENERGY_EFFICIENCY_CLASS = exports.ENERGY_EFFICIENCY_CLASS || (exports.ENERGY_EFFICIENCY_CLASS = {}));
exports.ALL_ENERGY_EFFICIENCY_CLASS = common_1.getAllEnumStatuses(ENERGY_EFFICIENCY_CLASS);
var ENERGY_SOURCE_TYPE;
(function (ENERGY_SOURCE_TYPE) {
    ENERGY_SOURCE_TYPE["NO_INFORMATION"] = "NO_INFORMATION";
    ENERGY_SOURCE_TYPE["GEOTHERMAL"] = "GEOTHERMAL";
    ENERGY_SOURCE_TYPE["SOLAR_HEATING"] = "SOLAR_HEATING";
    ENERGY_SOURCE_TYPE["PELLET_HEATING"] = "PELLET_HEATING";
    ENERGY_SOURCE_TYPE["GAS"] = "GAS";
    ENERGY_SOURCE_TYPE["OIL"] = "OIL";
    ENERGY_SOURCE_TYPE["DISTRICT_HEATING"] = "DISTRICT_HEATING";
    ENERGY_SOURCE_TYPE["ELECTRICITY"] = "ELECTRICITY";
    ENERGY_SOURCE_TYPE["COAL"] = "COAL";
    ENERGY_SOURCE_TYPE["ACID_GAS"] = "ACID_GAS";
    ENERGY_SOURCE_TYPE["SOUR_GAS"] = "SOUR_GAS";
    ENERGY_SOURCE_TYPE["LIQUID_GAS"] = "LIQUID_GAS";
    ENERGY_SOURCE_TYPE["STEAM_DISTRICT_HEATING"] = "STEAM_DISTRICT_HEATING";
    ENERGY_SOURCE_TYPE["WOOD"] = "WOOD";
    ENERGY_SOURCE_TYPE["WOOD_CHIPS"] = "WOOD_CHIPS";
    ENERGY_SOURCE_TYPE["COAL_COKE"] = "COAL_COKE";
    ENERGY_SOURCE_TYPE["LOCAL_HEATING"] = "LOCAL_HEATING";
    ENERGY_SOURCE_TYPE["HEAT_SUPPLY"] = "HEAT_SUPPLY";
    ENERGY_SOURCE_TYPE["BIO_ENERGY"] = "BIO_ENERGY";
    ENERGY_SOURCE_TYPE["HYDRO_ENERGY"] = "HYDRO_ENERGY";
    ENERGY_SOURCE_TYPE["ENVIRONMENTAL_THERMAL_ENERGY"] = "ENVIRONMENTAL_THERMAL_ENERGY";
    ENERGY_SOURCE_TYPE["COMBINED_HEAT_AND_POWER_FOSSIL_FUELS"] = "COMBINED_HEAT_AND_POWER_FOSSIL_FUELS";
    ENERGY_SOURCE_TYPE["COMBINED_HEAT_AND_POWER_RENEWABLE_ENERGY"] = "COMBINED_HEAT_AND_POWER_RENEWABLE_ENERGY";
    ENERGY_SOURCE_TYPE["COMBINED_HEAT_AND_POWER_REGENERATIVE_ENERGY"] = "COMBINED_HEAT_AND_POWER_REGENERATIVE_ENERGY";
    ENERGY_SOURCE_TYPE["COMBINED_HEAT_AND_POWER_BIO_ENERGY"] = "COMBINED_HEAT_AND_POWER_BIO_ENERGY";
})(ENERGY_SOURCE_TYPE = exports.ENERGY_SOURCE_TYPE || (exports.ENERGY_SOURCE_TYPE = {}));
exports.ALL_ENERGY_SOURCE_TYPE = common_1.getAllEnumStatuses(ENERGY_SOURCE_TYPE);
var CANDIDATE_FILTER_CATEGORY;
(function (CANDIDATE_FILTER_CATEGORY) {
    CANDIDATE_FILTER_CATEGORY["ALL"] = "ALL";
    CANDIDATE_FILTER_CATEGORY["TO_DO"] = "TO_DO";
    CANDIDATE_FILTER_CATEGORY["PENDING_CANDIDATE_INFORMATION"] = "PENDING_CANDIDATE_INFORMATION";
    CANDIDATE_FILTER_CATEGORY["INVITED"] = "INVITED";
    CANDIDATE_FILTER_CATEGORY["SCHEDULED"] = "SCHEDULED";
    CANDIDATE_FILTER_CATEGORY["APPLICATION_PENDING"] = "APPLICATION_PENDING";
    CANDIDATE_FILTER_CATEGORY["DATA_ROOM_SHARED"] = "DATA_ROOM_SHARED";
    CANDIDATE_FILTER_CATEGORY["APPLICANTS"] = "APPLICANTS";
    CANDIDATE_FILTER_CATEGORY["REJECTED"] = "REJECTED";
    CANDIDATE_FILTER_CATEGORY["UNREAD_MESSAGES"] = "UNREAD_MESSAGES";
})(CANDIDATE_FILTER_CATEGORY = exports.CANDIDATE_FILTER_CATEGORY || (exports.CANDIDATE_FILTER_CATEGORY = {}));
var CONSTRUCTION_PHASE;
(function (CONSTRUCTION_PHASE) {
    CONSTRUCTION_PHASE["PROJECTED"] = "PROJECTED";
    CONSTRUCTION_PHASE["UNDER_CONSTRUCTION"] = "UNDER_CONSTRUCTION";
    CONSTRUCTION_PHASE["COMPLETED"] = "COMPLETED";
    CONSTRUCTION_PHASE["NO_INFORMATION"] = "NO_INFORMATION";
})(CONSTRUCTION_PHASE = exports.CONSTRUCTION_PHASE || (exports.CONSTRUCTION_PHASE = {}));
exports.ALL_CONSTRUCTION_PHASE = common_1.getAllEnumStatuses(CONSTRUCTION_PHASE);
var LISTING_TYPE;
(function (LISTING_TYPE) {
    LISTING_TYPE["RENT_APARTMENT"] = "RENT_APARTMENT";
    LISTING_TYPE["SELL_APARTMENT"] = "SELL_APARTMENT";
    LISTING_TYPE["RENT_SHORT_TERM_APARTMENT"] = "RENT_SHORT_TERM_APARTMENT";
    LISTING_TYPE["RENT_HOUSE"] = "RENT_HOUSE";
    LISTING_TYPE["SELL_HOUSE"] = "SELL_HOUSE";
    LISTING_TYPE["RENT_PARKING"] = "RENT_PARKING";
    LISTING_TYPE["SELL_PARKING"] = "SELL_PARKING";
    LISTING_TYPE["RENT_RESIDENTIAL_LAND"] = "RENT_RESIDENTIAL_LAND";
    LISTING_TYPE["SELL_RESIDENTIAL_LAND"] = "SELL_RESIDENTIAL_LAND";
    LISTING_TYPE["RENT_OFFICE"] = "RENT_OFFICE";
    LISTING_TYPE["SELL_OFFICE"] = "SELL_OFFICE";
    LISTING_TYPE["RENT_STORE"] = "RENT_STORE";
    LISTING_TYPE["SELL_STORE"] = "SELL_STORE";
    LISTING_TYPE["RENT_GASTRONOMY"] = "RENT_GASTRONOMY";
    LISTING_TYPE["SELL_GASTRONOMY"] = "SELL_GASTRONOMY";
    LISTING_TYPE["RENT_INDUSTRY"] = "RENT_INDUSTRY";
    LISTING_TYPE["SELL_INDUSTRY"] = "SELL_INDUSTRY";
    LISTING_TYPE["RENT_SPECIAL_PURPOSE"] = "RENT_SPECIAL_PURPOSE";
    LISTING_TYPE["SELL_SPECIAL_PURPOSE"] = "SELL_SPECIAL_PURPOSE";
    LISTING_TYPE["RENT_COMMERCIAL_LAND"] = "RENT_COMMERCIAL_LAND";
    LISTING_TYPE["SELL_COMMERCIAL_LAND"] = "SELL_COMMERCIAL_LAND";
})(LISTING_TYPE = exports.LISTING_TYPE || (exports.LISTING_TYPE = {}));
exports.ALL_LISTING_TYPE = common_1.getAllEnumStatuses(LISTING_TYPE);
var TRANSACTION_TYPE;
(function (TRANSACTION_TYPE) {
    TRANSACTION_TYPE["RENT"] = "RENT";
    TRANSACTION_TYPE["SELL"] = "SELL";
})(TRANSACTION_TYPE = exports.TRANSACTION_TYPE || (exports.TRANSACTION_TYPE = {}));
exports.ALL_TRANSACTION_TYPE = common_1.getAllEnumStatuses(TRANSACTION_TYPE);
var SITE_DEVELOPMENT_TYPE;
(function (SITE_DEVELOPMENT_TYPE) {
    SITE_DEVELOPMENT_TYPE["DEVELOPED"] = "DEVELOPED";
    SITE_DEVELOPMENT_TYPE["NOT_DEVELOPED"] = "NOT_DEVELOPED";
    SITE_DEVELOPMENT_TYPE["DEVELOPED_PARTIALLY"] = "DEVELOPED_PARTIALLY";
    SITE_DEVELOPMENT_TYPE["NO_INFORMATION"] = "NO_INFORMATION";
})(SITE_DEVELOPMENT_TYPE = exports.SITE_DEVELOPMENT_TYPE || (exports.SITE_DEVELOPMENT_TYPE = {}));
exports.ALL_SITE_DEVELOPMENT_TYPE = common_1.getAllEnumStatuses(SITE_DEVELOPMENT_TYPE);
var SITE_CONSTRUCTIBLE_TYPE;
(function (SITE_CONSTRUCTIBLE_TYPE) {
    SITE_CONSTRUCTIBLE_TYPE["CONSTRUCTION_PLAN"] = "CONSTRUCTION_PLAN";
    SITE_CONSTRUCTIBLE_TYPE["NEIGHBOUR_CONSTRUCTION"] = "NEIGHBOUR_CONSTRUCTION";
    SITE_CONSTRUCTIBLE_TYPE["EXTERNAL_AREA"] = "EXTERNAL_AREA";
    SITE_CONSTRUCTIBLE_TYPE["NO_INFORMATION"] = "NO_INFORMATION";
})(SITE_CONSTRUCTIBLE_TYPE = exports.SITE_CONSTRUCTIBLE_TYPE || (exports.SITE_CONSTRUCTIBLE_TYPE = {}));
exports.ALL_SITE_CONSTRUCTIBLE_TYPE = common_1.getAllEnumStatuses(SITE_CONSTRUCTIBLE_TYPE);
var RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND;
(function (RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND) {
    RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND["FUTURE_DEVELOPMENT_LAND"] = "FUTURE_DEVELOPMENT_LAND";
    RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND["TWINHOUSE"] = "TWINHOUSE";
    RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND["SINGLE_FAMILY_HOUSE"] = "SINGLE_FAMILY_HOUSE";
    RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND["GARAGE"] = "GARAGE";
    RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND["GARDEN"] = "GARDEN";
    RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND["APARTMENT_BUILDING"] = "APARTMENT_BUILDING";
    RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND["ORCHARD"] = "ORCHARD";
    RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND["TERRACE_HOUSE"] = "TERRACE_HOUSE";
    RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND["PARKING_SPACE"] = "PARKING_SPACE";
    RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND["VILLA"] = "VILLA";
    RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND["FORREST"] = "FORREST";
    RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND["NO_DEVELOPMENT"] = "NO_DEVELOPMENT";
    RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND["NO_INFORMATION"] = "NO_INFORMATION";
})(RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND = exports.RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND || (exports.RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND = {}));
exports.ALL_RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND = common_1.getAllEnumStatuses(RECOMMENDED_USE_TYPE_RESIDENTIAL_LAND);
var RECOMMENDED_USE_TYPE_COMMERCIAL_LAND;
(function (RECOMMENDED_USE_TYPE_COMMERCIAL_LAND) {
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["FARMLAND"] = "FARMLAND";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["FUTURE_DEVELOPMENT_LAND"] = "FUTURE_DEVELOPMENT_LAND";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["MOORAGE"] = "MOORAGE";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["OFFICE"] = "OFFICE";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["CAMPING"] = "CAMPING";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["BIG_STORE"] = "BIG_STORE";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["LITTLE_STORE"] = "LITTLE_STORE";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["GARAGE"] = "GARAGE";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["GARDEN"] = "GARDEN";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["GASTRONOMY"] = "GASTRONOMY";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["BUSINESS"] = "BUSINESS";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["HOTEL"] = "HOTEL";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["INDUSTRY"] = "INDUSTRY";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["NO_DEVELOPMENT"] = "NO_DEVELOPMENT";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["SMALL_BUSINESS"] = "SMALL_BUSINESS";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["STOCK"] = "STOCK";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["ORCHARD"] = "ORCHARD";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["CAR_PARK"] = "CAR_PARK";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["PRODUCTION"] = "PRODUCTION";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["PARKING_SPACE"] = "PARKING_SPACE";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["FORREST"] = "FORREST";
    RECOMMENDED_USE_TYPE_COMMERCIAL_LAND["NO_INFORMATION"] = "NO_INFORMATION";
})(RECOMMENDED_USE_TYPE_COMMERCIAL_LAND = exports.RECOMMENDED_USE_TYPE_COMMERCIAL_LAND || (exports.RECOMMENDED_USE_TYPE_COMMERCIAL_LAND = {}));
exports.ALL_RECOMMENDED_USE_TYPE_COMMERCIAL_LAND = common_1.getAllEnumStatuses(RECOMMENDED_USE_TYPE_COMMERCIAL_LAND);
var MARKETING_TYPE;
(function (MARKETING_TYPE) {
    MARKETING_TYPE["RENT"] = "RENT";
    MARKETING_TYPE["LEASE"] = "LEASE";
    MARKETING_TYPE["LEASEHOLD"] = "LEASEHOLD";
    MARKETING_TYPE["PURCHASE"] = "PURCHASE";
    MARKETING_TYPE["BUY"] = "BUY";
})(MARKETING_TYPE = exports.MARKETING_TYPE || (exports.MARKETING_TYPE = {}));
exports.ALL_MARKETING_TYPE = common_1.getAllEnumStatuses(MARKETING_TYPE);
var RENTAL_INTERVAL_TYPE_OFFICE;
(function (RENTAL_INTERVAL_TYPE_OFFICE) {
    RENTAL_INTERVAL_TYPE_OFFICE["WEEKLY"] = "WEEKLY";
    RENTAL_INTERVAL_TYPE_OFFICE["MONTHLY"] = "MONTHLY";
    RENTAL_INTERVAL_TYPE_OFFICE["YEARLY"] = "YEARLY";
    RENTAL_INTERVAL_TYPE_OFFICE["LONG_TERM"] = "LONG_TERM";
})(RENTAL_INTERVAL_TYPE_OFFICE = exports.RENTAL_INTERVAL_TYPE_OFFICE || (exports.RENTAL_INTERVAL_TYPE_OFFICE = {}));
exports.ALL_RENTAL_INTERVAL_TYPE_OFFICE = common_1.getAllEnumStatuses(RENTAL_INTERVAL_TYPE_OFFICE);
var STORE_SUPPLY_TYPE;
(function (STORE_SUPPLY_TYPE) {
    STORE_SUPPLY_TYPE["DIRECT_APPROACH"] = "DIRECT_APPROACH";
    STORE_SUPPLY_TYPE["NO_DIRECT_APPROACH"] = "NO_DIRECT_APPROACH";
    STORE_SUPPLY_TYPE["CAR_APPROACH"] = "CAR_APPROACH";
    STORE_SUPPLY_TYPE["APPROACH_TO_THE_FRONT"] = "APPROACH_TO_THE_FRONT";
    STORE_SUPPLY_TYPE["APPROACH_TO_THE_BACK"] = "APPROACH_TO_THE_BACK";
    STORE_SUPPLY_TYPE["FULL_TIME"] = "FULL_TIME";
    STORE_SUPPLY_TYPE["FORENOON"] = "FORENOON";
    STORE_SUPPLY_TYPE["NO_INFORMATION"] = "NO_INFORMATION";
})(STORE_SUPPLY_TYPE = exports.STORE_SUPPLY_TYPE || (exports.STORE_SUPPLY_TYPE = {}));
exports.ALL_STORE_SUPPLY_TYPE = common_1.getAllEnumStatuses(STORE_SUPPLY_TYPE);
var YES_NO_BYAPPOINTMENT;
(function (YES_NO_BYAPPOINTMENT) {
    YES_NO_BYAPPOINTMENT["YES"] = "YES";
    YES_NO_BYAPPOINTMENT["NO"] = "NO";
    YES_NO_BYAPPOINTMENT["BY_APPOINTMENT"] = "BY_APPOINTMENT";
    YES_NO_BYAPPOINTMENT["NO_INFORMATION"] = "NO_INFORMATION";
})(YES_NO_BYAPPOINTMENT = exports.YES_NO_BYAPPOINTMENT || (exports.YES_NO_BYAPPOINTMENT = {}));
exports.ALL_YES_NO_BYAPPOINTMENT = common_1.getAllEnumStatuses(YES_NO_BYAPPOINTMENT);
var RENT_TYPE;
(function (RENT_TYPE) {
    RENT_TYPE["RENT_PER_MONTH"] = "RENT_PER_MONTH";
    RENT_TYPE["RENT_PER_SQM"] = "RENT_PER_SQM";
})(RENT_TYPE = exports.RENT_TYPE || (exports.RENT_TYPE = {}));
exports.ALL_RENT_TYPE = common_1.getAllEnumStatuses(RENT_TYPE);
