// @everreal/er-common-helpers/lib/helpers/constants/mediaTypes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var common_1 = require("./common");
var MEDIA_TYPE_HEADERS;
(function (MEDIA_TYPE_HEADERS) {
    MEDIA_TYPE_HEADERS["TEXT_PLAIN"] = "text/plain";
    MEDIA_TYPE_HEADERS["TEXT_CSV"] = "text/csv";
    MEDIA_TYPE_HEADERS["TEXT_HTML"] = "text/html";
    MEDIA_TYPE_HEADERS["TEXT_CSS"] = "text/css";
    MEDIA_TYPE_HEADERS["IMAGE_SVG"] = "image/svg+xml";
    MEDIA_TYPE_HEADERS["IMAGE_PNG"] = "image/png";
    MEDIA_TYPE_HEADERS["IMAGE_JPEG"] = "image/jpeg";
    MEDIA_TYPE_HEADERS["IMAGE_GIF"] = "image/gif";
    MEDIA_TYPE_HEADERS["AUDIO_MPEG"] = "audio/mpeg";
    MEDIA_TYPE_HEADERS["AUDIO_OGG"] = "audio/ogg";
    MEDIA_TYPE_HEADERS["APPLICATION_OCTET_STREAM"] = "application/octet-stream";
    MEDIA_TYPE_HEADERS["APPLICATION_PDF"] = "application/pdf";
    MEDIA_TYPE_HEADERS["APPLICATION_XML"] = "application/xml";
    MEDIA_TYPE_HEADERS["APPLICATION_JAVASCRIPT"] = "application/javascript";
    MEDIA_TYPE_HEADERS["APPLICATION_JSON"] = "application/json";
    MEDIA_TYPE_HEADERS["APPLICATION_FORM_URLENCODED"] = "application/x-www-formurlencoded-urlencoded";
    MEDIA_TYPE_HEADERS["APPLICATION_ZIP"] = "application/zip";
    MEDIA_TYPE_HEADERS["APPLICATION_SQL"] = "application/sql";
    MEDIA_TYPE_HEADERS["APPLICATION_GRAPHQL"] = "application/graphql";
    MEDIA_TYPE_HEADERS["APPLICATION_DOC"] = "application/msword";
    MEDIA_TYPE_HEADERS["APPLICATION_DOCX"] = "application/vnd.openxmlformats-officedocument.wordprocessingml.document";
    MEDIA_TYPE_HEADERS["APPLICATION_DOTX"] = "application/vnd.openxmlformats-officedocument.wordprocessingml.template";
    MEDIA_TYPE_HEADERS["APPLICATION_XLS"] = "application/vnd.ms-excel";
    MEDIA_TYPE_HEADERS["APPLICATION_XLSX"] = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet";
    MEDIA_TYPE_HEADERS["APPLICATION_XLTX"] = "application/vnd.openxmlformats-officedocument.spreadsheetml.template";
    MEDIA_TYPE_HEADERS["APPLICATION_PPT"] = "application/vnd.ms-powerpoint";
    MEDIA_TYPE_HEADERS["APPLICATION_PPTX"] = "application/vnd.openxmlformats-officedocument.presentationml.presentation";
    MEDIA_TYPE_HEADERS["APPLICATION_POTX"] = "application/vnd.openxmlformats-officedocument.presentationml.template";
    MEDIA_TYPE_HEADERS["APPLICATION_PPSX"] = "application/vnd.openxmlformats-officedocument.presentationml.slideshow";
    MEDIA_TYPE_HEADERS["APPLICATION_MDB"] = "application/vnd.ms-access";
    MEDIA_TYPE_HEADERS["MULTIPART_FORM_DATA"] = "multipart/form-data";
})(MEDIA_TYPE_HEADERS = exports.MEDIA_TYPE_HEADERS || (exports.MEDIA_TYPE_HEADERS = {}));
exports.ALL_MEDIA_TYPE_HEADERS = common_1.getAllEnumStatuses(MEDIA_TYPE_HEADERS);
