// @everreal/er-common-helpers/lib/helpers/constants/moveIn.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var common_1 = require("./common");
exports.FLOOR_TYPE = {
    PARQUET: 'PARQUET',
    CARPET: 'CARPET',
    PRE_BUILT_PARQUET: 'PRE_BUILT_PARQUET',
    LAMINTE: 'LAMINTE',
    TILES: 'TILES',
    PVC: 'PVC',
    OTHER: 'OTHER',
};
exports.ALL_FLOOR_TYPE = common_1.getAllStatuses(exports.FLOOR_TYPE);
exports.WALL_TYPE = {
    PAINTED_WHITE: 'PAINTED_WHITE',
    WALLPAPER: 'WALLPAPER',
    TILES: 'TILES',
    OTHER: 'OTHER',
};
exports.ALL_WALL_TYPE = common_1.getAllStatuses(exports.WALL_TYPE);
exports.CEILING_TYPE = {
    PAINTED_WHITE: 'PAINTED_WHITE',
    WALLPAPER: 'WALLPAPER',
    TILES: 'TILES',
    OTHER: 'OTHER',
};
exports.ALL_CEILING_TYPE = common_1.getAllStatuses(exports.CEILING_TYPE);
exports.METER_TYPE = {
    HEATING: 'HEATING',
    COLD_WATER: 'COLD_WATER',
    WARM_WATER: 'WARM_WATER',
    ELECTRICITY: 'ELECTRICITY',
    OTHER: 'OTHER',
};
exports.ALL_METER_TYPE = common_1.getAllStatuses(exports.METER_TYPE);
exports.KEY_TYPE = {
    ENTRY_DOOR: 'ENTRY_DOOR',
    APARTMENT_DOOR: 'APARTMENT_DOOR',
    MAILBOX: 'MAILBOX',
    BASEMENT: 'BASEMENT',
    LAUNDRY_ROOM: 'LAUNDRY_ROOM',
    ATTIC: 'ATTIC',
    OTHER: 'OTHER',
};
exports.ALL_KEY_TYPE = common_1.getAllStatuses(exports.KEY_TYPE);
var PROTOCOL_STATUS;
(function (PROTOCOL_STATUS) {
    PROTOCOL_STATUS["__DEFAULT"] = "UNSYNCED";
    PROTOCOL_STATUS["UNSYNCED"] = "UNSYNCED";
    PROTOCOL_STATUS["SYNCING"] = "SYNCING";
    PROTOCOL_STATUS["SYNCED"] = "SYNCED";
})(PROTOCOL_STATUS = exports.PROTOCOL_STATUS || (exports.PROTOCOL_STATUS = {}));
exports.ALL_PROTOCOL_STATUS = common_1.getAllEnumStatuses(PROTOCOL_STATUS);
var PROTOCOL_TYPE;
(function (PROTOCOL_TYPE) {
    PROTOCOL_TYPE["MOVE_IN"] = "MOVE_IN";
    PROTOCOL_TYPE["MOVE_OUT"] = "MOVE_OUT";
})(PROTOCOL_TYPE = exports.PROTOCOL_TYPE || (exports.PROTOCOL_TYPE = {}));
exports.ALL_PROTOCOL_TYPE = common_1.getAllEnumStatuses(PROTOCOL_TYPE);
var MOVEIN_DEPOSIT_TRANSFER_TYPE;
(function (MOVEIN_DEPOSIT_TRANSFER_TYPE) {
    MOVEIN_DEPOSIT_TRANSFER_TYPE["CASH_TRANSFER"] = "CASH_TRANSFER";
    MOVEIN_DEPOSIT_TRANSFER_TYPE["WIRE_TRANSFER"] = "WIRE_TRANSFER";
})(MOVEIN_DEPOSIT_TRANSFER_TYPE = exports.MOVEIN_DEPOSIT_TRANSFER_TYPE || (exports.MOVEIN_DEPOSIT_TRANSFER_TYPE = {}));
exports.ALL_MOVEIN_DEPOSIT_TRANSFER_TYPE = common_1.getAllEnumStatuses(MOVEIN_DEPOSIT_TRANSFER_TYPE);
