// @everreal/er-common-helpers/lib/helpers/constants/properties.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var common_1 = require("./common");
var PROPERTY_CATEGORY;
(function (PROPERTY_CATEGORY) {
    PROPERTY_CATEGORY["RESIDENTIAL"] = "RESIDENTIAL";
    PROPERTY_CATEGORY["COMMERCIAL"] = "COMMERCIAL";
    PROPERTY_CATEGORY["RESIDENTIAL_AND_COMMERCIAL"] = "RESIDENTIAL_AND_COMMERCIAL";
})(PROPERTY_CATEGORY = exports.PROPERTY_CATEGORY || (exports.PROPERTY_CATEGORY = {}));
exports.ALL_PROPERTY_CATEGORY = common_1.getAllEnumStatuses(PROPERTY_CATEGORY);
var PROPERTY_TYPE;
(function (PROPERTY_TYPE) {
    PROPERTY_TYPE["BUILDING"] = "BUILDING";
    PROPERTY_TYPE["LAND"] = "LAND";
})(PROPERTY_TYPE = exports.PROPERTY_TYPE || (exports.PROPERTY_TYPE = {}));
exports.ALL_PROPERTY_TYPE = common_1.getAllEnumStatuses(PROPERTY_TYPE);
var COMMERCIAL_LAND_SUBTYPE;
(function (COMMERCIAL_LAND_SUBTYPE) {
    COMMERCIAL_LAND_SUBTYPE["LEISURE"] = "LEISURE";
    COMMERCIAL_LAND_SUBTYPE["TRADE"] = "TRADE";
    COMMERCIAL_LAND_SUBTYPE["AGRICULTURE_FORESTRY"] = "AGRICULTURE_FORESTRY";
})(COMMERCIAL_LAND_SUBTYPE = exports.COMMERCIAL_LAND_SUBTYPE || (exports.COMMERCIAL_LAND_SUBTYPE = {}));
exports.ALL_COMMERCIAL_LAND_SUBTYPE = common_1.getAllEnumStatuses(COMMERCIAL_LAND_SUBTYPE);
