// @everreal/er-common-helpers/lib/helpers/constants/units.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var common_1 = require("./common");
var UNIT_CATEGORY;
(function (UNIT_CATEGORY) {
    UNIT_CATEGORY["RESIDENTIAL"] = "RESIDENTIAL";
    UNIT_CATEGORY["COMMERCIAL"] = "COMMERCIAL";
})(UNIT_CATEGORY = exports.UNIT_CATEGORY || (exports.UNIT_CATEGORY = {}));
exports.ALL_UNIT_CATEGORY = common_1.getAllEnumStatuses(UNIT_CATEGORY);
var UNIT_TYPE;
(function (UNIT_TYPE) {
    UNIT_TYPE["APARTMENT"] = "APARTMENT";
    UNIT_TYPE["HOUSE"] = "HOUSE";
    UNIT_TYPE["PARKING"] = "PARKING";
    UNIT_TYPE["OFFICE"] = "OFFICE";
    UNIT_TYPE["STORE"] = "STORE";
    UNIT_TYPE["GASTRONOMY"] = "GASTRONOMY";
    UNIT_TYPE["INDUSTRY"] = "INDUSTRY";
    UNIT_TYPE["SPECIAL_PURPOSE"] = "SPECIAL_PURPOSE";
    UNIT_TYPE["LAND"] = "LAND";
})(UNIT_TYPE = exports.UNIT_TYPE || (exports.UNIT_TYPE = {}));
exports.ALL_UNIT_TYPE = common_1.getAllEnumStatuses(UNIT_TYPE);
var UNIT_APARTMENT_SUBTYPE;
(function (UNIT_APARTMENT_SUBTYPE) {
    UNIT_APARTMENT_SUBTYPE["LOFT"] = "LOFT";
    UNIT_APARTMENT_SUBTYPE["MAISONETTE"] = "MAISONETTE";
    UNIT_APARTMENT_SUBTYPE["PENTHOUSE"] = "PENTHOUSE";
    UNIT_APARTMENT_SUBTYPE["TERRACE_APARTMENT"] = "TERRACE_APARTMENT";
    UNIT_APARTMENT_SUBTYPE["GROUND_FLOOR"] = "GROUND_FLOOR";
    UNIT_APARTMENT_SUBTYPE["APARTMENT"] = "APARTMENT";
    UNIT_APARTMENT_SUBTYPE["MEZZANINE"] = "MEZZANINE";
    UNIT_APARTMENT_SUBTYPE["BASEMENT"] = "BASEMENT";
    UNIT_APARTMENT_SUBTYPE["ATTIC_FLOOR"] = "ATTIC_FLOOR";
    UNIT_APARTMENT_SUBTYPE["HOUSE"] = "HOUSE";
    UNIT_APARTMENT_SUBTYPE["ROOM"] = "ROOM";
    UNIT_APARTMENT_SUBTYPE["STUDIO"] = "STUDIO";
    UNIT_APARTMENT_SUBTYPE["NO_INFORMATION"] = "NO_INFORMATION";
})(UNIT_APARTMENT_SUBTYPE = exports.UNIT_APARTMENT_SUBTYPE || (exports.UNIT_APARTMENT_SUBTYPE = {}));
exports.ALL_UNIT_APARTMENT_SUBTYPE = common_1.getAllEnumStatuses(UNIT_APARTMENT_SUBTYPE);
var UNIT_HOUSE_SUBTYPE;
(function (UNIT_HOUSE_SUBTYPE) {
    UNIT_HOUSE_SUBTYPE["SINGLE_FAMILY_HOUSE"] = "SINGLE_FAMILY_HOUSE";
    UNIT_HOUSE_SUBTYPE["MID_TERRACE_HOUSE"] = "MID_TERRACE_HOUSE";
    UNIT_HOUSE_SUBTYPE["END_TERRACE_HOUSE"] = "END_TERRACE_HOUSE";
    UNIT_HOUSE_SUBTYPE["MULTI_FAMILY_HOUSE"] = "MULTI_FAMILY_HOUSE";
    UNIT_HOUSE_SUBTYPE["BUNGALOW"] = "BUNGALOW";
    UNIT_HOUSE_SUBTYPE["FARMHOUSE"] = "FARMHOUSE";
    UNIT_HOUSE_SUBTYPE["SEMI_DETACHED_HOUSE"] = "SEMI_DETACHED_HOUSE";
    UNIT_HOUSE_SUBTYPE["MANSION"] = "MANSION";
    UNIT_HOUSE_SUBTYPE["TOWN_HOUSE"] = "TOWN_HOUSE";
    UNIT_HOUSE_SUBTYPE["SPECIAL_REAL_ESTATE"] = "SPECIAL_REAL_ESTATE";
    UNIT_HOUSE_SUBTYPE["NO_INFORMATION"] = "NO_INFORMATION";
})(UNIT_HOUSE_SUBTYPE = exports.UNIT_HOUSE_SUBTYPE || (exports.UNIT_HOUSE_SUBTYPE = {}));
exports.ALL_UNIT_HOUSE_SUBTYPE = common_1.getAllEnumStatuses(UNIT_HOUSE_SUBTYPE);
var UNIT_PARKING_SUBTYPE;
(function (UNIT_PARKING_SUBTYPE) {
    UNIT_PARKING_SUBTYPE["GARAGE"] = "GARAGE";
    UNIT_PARKING_SUBTYPE["OUTSIDE_PARKING_SPOT"] = "OUTSIDE_PARKING_SPOT";
    UNIT_PARKING_SUBTYPE["CARPORT"] = "CARPORT";
    UNIT_PARKING_SUBTYPE["DUPLEX"] = "DUPLEX";
    UNIT_PARKING_SUBTYPE["PARKING_GARAGE"] = "PARKING_GARAGE";
    UNIT_PARKING_SUBTYPE["UNDERGROUND_PARKING"] = "UNDERGROUND_PARKING";
    UNIT_PARKING_SUBTYPE["NO_INFORMATION"] = "NO_INFORMATION";
})(UNIT_PARKING_SUBTYPE = exports.UNIT_PARKING_SUBTYPE || (exports.UNIT_PARKING_SUBTYPE = {}));
exports.ALL_UNIT_PARKING_SUBTYPE = common_1.getAllEnumStatuses(UNIT_PARKING_SUBTYPE);
var UNIT_OFFICE_SUBTYPE;
(function (UNIT_OFFICE_SUBTYPE) {
    UNIT_OFFICE_SUBTYPE["LOFT"] = "LOFT";
    UNIT_OFFICE_SUBTYPE["STUDIO"] = "STUDIO";
    UNIT_OFFICE_SUBTYPE["OFFICE"] = "OFFICE";
    UNIT_OFFICE_SUBTYPE["OFFICE_FLOOR"] = "OFFICE_FLOOR";
    UNIT_OFFICE_SUBTYPE["OFFICE_CENTRE"] = "OFFICE_CENTRE";
    UNIT_OFFICE_SUBTYPE["OFFICE_STORAGE_BUILDING"] = "OFFICE_STORAGE_BUILDING";
    UNIT_OFFICE_SUBTYPE["SURGERY"] = "SURGERY";
    UNIT_OFFICE_SUBTYPE["SURGERY_FLOOR"] = "SURGERY_FLOOR";
    UNIT_OFFICE_SUBTYPE["SURGERY_BUILDING"] = "SURGERY_BUILDING";
    UNIT_OFFICE_SUBTYPE["COMMERCIAL_CENTRE"] = "COMMERCIAL_CENTRE";
    UNIT_OFFICE_SUBTYPE["LIVING_AND_COMMERCIAL_BUILDING"] = "LIVING_AND_COMMERCIAL_BUILDING";
    UNIT_OFFICE_SUBTYPE["OFFICE_AND_COMMERCIAL_BUILDING"] = "OFFICE_AND_COMMERCIAL_BUILDING";
})(UNIT_OFFICE_SUBTYPE = exports.UNIT_OFFICE_SUBTYPE || (exports.UNIT_OFFICE_SUBTYPE = {}));
exports.ALL_UNIT_OFFICE_SUBTYPE = common_1.getAllEnumStatuses(UNIT_OFFICE_SUBTYPE);
var UNIT_STORE_SUBTYPE;
(function (UNIT_STORE_SUBTYPE) {
    UNIT_STORE_SUBTYPE["SHOWROOM_SPACE"] = "SHOWROOM_SPACE";
    UNIT_STORE_SUBTYPE["SHOPPING_CENTRE"] = "SHOPPING_CENTRE";
    UNIT_STORE_SUBTYPE["FACTORY_OUTLET"] = "FACTORY_OUTLET";
    UNIT_STORE_SUBTYPE["DEPARTMENT_STORE"] = "DEPARTMENT_STORE";
    UNIT_STORE_SUBTYPE["KIOSK"] = "KIOSK";
    UNIT_STORE_SUBTYPE["STORE"] = "STORE";
    UNIT_STORE_SUBTYPE["SELF_SERVICE_MARKET"] = "SELF_SERVICE_MARKET";
    UNIT_STORE_SUBTYPE["SALES_AREA"] = "SALES_AREA";
    UNIT_STORE_SUBTYPE["SALES_HALL"] = "SALES_HALL";
})(UNIT_STORE_SUBTYPE = exports.UNIT_STORE_SUBTYPE || (exports.UNIT_STORE_SUBTYPE = {}));
exports.ALL_UNIT_STORE_SUBTYPE = common_1.getAllEnumStatuses(UNIT_STORE_SUBTYPE);
var UNIT_GASTRONOMY_SUBTYPE;
(function (UNIT_GASTRONOMY_SUBTYPE) {
    UNIT_GASTRONOMY_SUBTYPE["BAR_LOUNGE"] = "BAR_LOUNGE";
    UNIT_GASTRONOMY_SUBTYPE["CAFE"] = "CAFE";
    UNIT_GASTRONOMY_SUBTYPE["CLUB_DISCO"] = "CLUB_DISCO";
    UNIT_GASTRONOMY_SUBTYPE["GUESTS_HOUSE"] = "GUESTS_HOUSE";
    UNIT_GASTRONOMY_SUBTYPE["TAVERN"] = "TAVERN";
    UNIT_GASTRONOMY_SUBTYPE["HOTEL"] = "HOTEL";
    UNIT_GASTRONOMY_SUBTYPE["HOTEL_RESIDENCE"] = "HOTEL_RESIDENCE";
    UNIT_GASTRONOMY_SUBTYPE["HOTEL_GARNI"] = "HOTEL_GARNI";
    UNIT_GASTRONOMY_SUBTYPE["PENSION"] = "PENSION";
    UNIT_GASTRONOMY_SUBTYPE["RESTAURANT"] = "RESTAURANT";
    UNIT_GASTRONOMY_SUBTYPE["BUNGALOW"] = "BUNGALOW";
})(UNIT_GASTRONOMY_SUBTYPE = exports.UNIT_GASTRONOMY_SUBTYPE || (exports.UNIT_GASTRONOMY_SUBTYPE = {}));
exports.ALL_UNIT_GASTRONOMY_SUBTYPE = common_1.getAllEnumStatuses(UNIT_GASTRONOMY_SUBTYPE);
var UNIT_INDUSTRY_SUBTYPE;
(function (UNIT_INDUSTRY_SUBTYPE) {
    UNIT_INDUSTRY_SUBTYPE["SHOWROOM_SPACE"] = "SHOWROOM_SPACE";
    UNIT_INDUSTRY_SUBTYPE["HALL"] = "HALL";
    UNIT_INDUSTRY_SUBTYPE["HIGH_LACK_STORAGE"] = "HIGH_LACK_STORAGE";
    UNIT_INDUSTRY_SUBTYPE["INDUSTRY_HALL"] = "INDUSTRY_HALL";
    UNIT_INDUSTRY_SUBTYPE["COLD_STORAGE"] = "COLD_STORAGE";
    UNIT_INDUSTRY_SUBTYPE["MULTIDECK_CABINET_STORAGE"] = "MULTIDECK_CABINET_STORAGE";
    UNIT_INDUSTRY_SUBTYPE["STORAGE_WITH_OPEN_AREA"] = "STORAGE_WITH_OPEN_AREA";
    UNIT_INDUSTRY_SUBTYPE["STORAGE_AREA"] = "STORAGE_AREA";
    UNIT_INDUSTRY_SUBTYPE["STORAGE_HALL"] = "STORAGE_HALL";
    UNIT_INDUSTRY_SUBTYPE["SERVICE_AREA"] = "SERVICE_AREA";
    UNIT_INDUSTRY_SUBTYPE["SHIPPING_STORAGE"] = "SHIPPING_STORAGE";
    UNIT_INDUSTRY_SUBTYPE["REPAIR_SHOP"] = "REPAIR_SHOP";
})(UNIT_INDUSTRY_SUBTYPE = exports.UNIT_INDUSTRY_SUBTYPE || (exports.UNIT_INDUSTRY_SUBTYPE = {}));
exports.ALL_UNIT_INDUSTRY_SUBTYPE = common_1.getAllEnumStatuses(UNIT_INDUSTRY_SUBTYPE);
var UNIT_SPECIAL_PURPOSE_SUBTYPE;
(function (UNIT_SPECIAL_PURPOSE_SUBTYPE) {
    UNIT_SPECIAL_PURPOSE_SUBTYPE["RESIDENCE"] = "RESIDENCE";
    UNIT_SPECIAL_PURPOSE_SUBTYPE["FARM"] = "FARM";
    UNIT_SPECIAL_PURPOSE_SUBTYPE["HORSE_FARM"] = "HORSE_FARM";
    UNIT_SPECIAL_PURPOSE_SUBTYPE["VINEYARD"] = "VINEYARD";
    UNIT_SPECIAL_PURPOSE_SUBTYPE["REPAIR_SHOP"] = "REPAIR_SHOP";
    UNIT_SPECIAL_PURPOSE_SUBTYPE["LEISURE_FACILITY"] = "LEISURE_FACILITY";
    UNIT_SPECIAL_PURPOSE_SUBTYPE["SPECIAL_ESTATE"] = "SPECIAL_ESTATE";
    UNIT_SPECIAL_PURPOSE_SUBTYPE["COMMERCIAL_CENTRE"] = "COMMERCIAL_CENTRE";
    UNIT_SPECIAL_PURPOSE_SUBTYPE["INDUSTRIAL_AREA"] = "INDUSTRIAL_AREA";
})(UNIT_SPECIAL_PURPOSE_SUBTYPE = exports.UNIT_SPECIAL_PURPOSE_SUBTYPE || (exports.UNIT_SPECIAL_PURPOSE_SUBTYPE = {}));
exports.ALL_UNIT_SPECIAL_PURPOSE_SUBTYPE = common_1.getAllEnumStatuses(UNIT_SPECIAL_PURPOSE_SUBTYPE);
var UNIT_LEASING_STATUSES;
(function (UNIT_LEASING_STATUSES) {
    UNIT_LEASING_STATUSES["LEASED"] = "LEASED";
    UNIT_LEASING_STATUSES["TERMINATED"] = "TERMINATED";
    UNIT_LEASING_STATUSES["VACANT"] = "VACANT";
    UNIT_LEASING_STATUSES["FUTURE_LEASED"] = "FUTURE_LEASED";
})(UNIT_LEASING_STATUSES = exports.UNIT_LEASING_STATUSES || (exports.UNIT_LEASING_STATUSES = {}));
exports.ALL_UNIT_LEASING_STATUSES = common_1.getAllEnumStatuses(UNIT_LEASING_STATUSES);
var UNIT_STORE_POSITION_TYPE;
(function (UNIT_STORE_POSITION_TYPE) {
    UNIT_STORE_POSITION_TYPE["CLASSIFICATION_A"] = "CLASSIFICATION_A";
    UNIT_STORE_POSITION_TYPE["CLASSIFICATION_B"] = "CLASSIFICATION_B";
    UNIT_STORE_POSITION_TYPE["SHOPPING_CENTRE"] = "SHOPPING_CENTRE";
    UNIT_STORE_POSITION_TYPE["NO_INFORMATION"] = "NO_INFORMATION";
})(UNIT_STORE_POSITION_TYPE = exports.UNIT_STORE_POSITION_TYPE || (exports.UNIT_STORE_POSITION_TYPE = {}));
exports.ALL_UNIT_STORE_POSITION_TYPE = common_1.getAllEnumStatuses(UNIT_STORE_POSITION_TYPE);
var COMMERCIAL_UNIT_FLOORTYPE;
(function (COMMERCIAL_UNIT_FLOORTYPE) {
    COMMERCIAL_UNIT_FLOORTYPE["CONCRETE"] = "CONCRETE";
    COMMERCIAL_UNIT_FLOORTYPE["EPOXY_RESIN"] = "EPOXY_RESIN";
    COMMERCIAL_UNIT_FLOORTYPE["TILES"] = "TILES";
    COMMERCIAL_UNIT_FLOORTYPE["PLANKS"] = "PLANKS";
    COMMERCIAL_UNIT_FLOORTYPE["LAMINATE"] = "LAMINATE";
    COMMERCIAL_UNIT_FLOORTYPE["PARQUET"] = "PARQUET";
    COMMERCIAL_UNIT_FLOORTYPE["PVC"] = "PVC";
    COMMERCIAL_UNIT_FLOORTYPE["CARPET"] = "CARPET";
    COMMERCIAL_UNIT_FLOORTYPE["ANTISTATIC_FLOOR"] = "ANTISTATIC_FLOOR";
    COMMERCIAL_UNIT_FLOORTYPE["OFFICE_CARPET"] = "OFFICE_CARPET";
    COMMERCIAL_UNIT_FLOORTYPE["STONE"] = "STONE";
    COMMERCIAL_UNIT_FLOORTYPE["CUSTOMIZABLE"] = "CUSTOMIZABLE";
    COMMERCIAL_UNIT_FLOORTYPE["WITHOUT"] = "WITHOUT";
    COMMERCIAL_UNIT_FLOORTYPE["NO_INFORMATION"] = "NO_INFORMATION";
})(COMMERCIAL_UNIT_FLOORTYPE = exports.COMMERCIAL_UNIT_FLOORTYPE || (exports.COMMERCIAL_UNIT_FLOORTYPE = {}));
exports.ALL_COMMERCIAL_UNIT_FLOORTYPE = common_1.getAllEnumStatuses(COMMERCIAL_UNIT_FLOORTYPE);
